### ISSUE TITLE
InkWell does not show through opaque content

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#57260]
Internal: b/156451097

Yep, title is a bit weird. However, the code makes sense, I promise.

```dart
class MyWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Card(
      child: InkWell(
        onTap: () {},
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            Container(
              width: 200,
              height: 100,
              child: Center(
                child: Text('Title'),
              ),
            ),
            Container(
              width: 200,
              height: 200,
              color: Colors.red,
            ),
            Container(
              width: 200,
              height: 100,
              child: Center(
                child: RaisedButton(child: Icon(Icons.arrow_forward),
                                   onPressed: () {},),
              ),
            ),
          ],
        ),
      ),
    );
  }
}
```

This makes the container clickable but the ripple effect does not show. This is actually an a11y bug (I was not aware that ripple had an a11y component to it).

If we try the `Stack` workaround, it works but now we can't tap on anything that's under the InkWell including the button. Now the user is left with trying to lay all these components manually.

I also suggested the workaround of drawing the container with 0.8 opacity. For instance, do this for images:

```dart
child: Image.network(
                  'some image',
                  color: Color.fromRGBO(255, 255, 255, 0.8),
                  colorBlendMode: BlendMode.modulate,
),
```

This feels like a very poor workaround. It also does not work well because by the time the image shows up as designed, the ripple that shows through is pretty weak.

Is there a way to ensure that InkWell draws on top of all of its children... unless they themselves are clickable?

**Comment by HansMuller** [comment]
CC @dkwingsmt

**Comment by dkwingsmt** [comment]
Here's the internal issue http://go/recall/-/aaaaaabFQoREREyvrsCO/b0MVbZoNuug3rryTddPanT 
(Google internal link)

**Comment by Hixie** [comment]
InkDecoration is the normal solution for this, FWIW.

**Comment by dkwingsmt** [comment]
(Actually it's `Ink`)
@mehmetf Can you let them know if this can be solved by placing the image in `Ink.image`?

**Comment by mehmetf** [comment]
I was not aware of `InkDecoration` and `Ink`. Let me suggest that. I can see this being difficult if the image or other opaque content is several widgets down the tree but it should be doable.

**Comment by mehmetf** [comment]
We have an internal thread going on about this. Lowering priority in the meantime.

**Comment by TahaTesser** [comment]
Hi @mehmetf 
Is this still a valid issue

Maybe I am missing something but I can see the ripple effect

<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: Center(
        child: MyWidget(),
      ),
      floatingActionButton: FloatingActionButton(
        child: Icon(Icons.add),
        onPressed: () {},
      ),
    );
  }
}

class MyWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Card(
      child: InkWell(
        onTap: () {},
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            Container(
              width: 200,
              height: 100,
              child: Center(
                child: Text('Title'),
              ),
            ),
            Container(
              width: 200,
              height: 200,
              color: Colors.red,
            ),
            Container(
              width: 200,
              height: 100,
              child: Center(
                child: RaisedButton(
                  child: Icon(Icons.arrow_forward),
                  onPressed: () {},
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

```

</details> 


<details> 
<summary>flutter doctor -v</summary> 

```
[✓] Flutter (Channel master, 1.22.0-2.0.pre.162, on Microsoft Windows [Version 10.0.19041.450], locale en-US)
    • Flutter version 1.22.0-2.0.pre.162 at C:\Code\flutter_master
    • Framework revision bcfad2068e (60 minutes ago), 2020-09-01 08:34:25 -0400
    • Engine revision 165abef0a2
    • Dart version 2.10.0 (build 2.10.0-77.0.dev)


[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Code\sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = C:\Code\sdk
    • Java binary at: C:\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.7.2)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.7.30413.136
    • Windows 10 SDK version 10.0.18362.0

[✓] Android Studio (version 4.0)
    • Android Studio at C:\Code\android-studio
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] VS Code (version 1.48.2)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.13.2

[✓] Connected device (4 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86    • Android 10 (API 29) (emulator)
    • Windows (desktop)                  • windows       • windows-x64    • Microsoft Windows [Version 10.0.19041.450]
    • Web Server (web)                   • web-server    • web-javascript • Flutter Tools
    • Chrome (web)                       • chrome        • web-javascript • Google Chrome 84.0.4147.135

• No issues found!
```

</details>

**Comment by mehmetf** [comment]
@TahaTesser Try tapping on the red container (using your code sample). The ripple won't show. It does show on the other pieces (title and button).

**Comment by TahaTesser** [comment]
Thanks,  definitely show over the red container

**Comment by enetor** [comment]
Internal bug 156451097 has been closed.

**Comment by dkwingsmt** [comment]
Closing this issue following the closure of internal issue.
